daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -127,3 +127,15 @@ message NodeRollback { // *since version 14*
 }
 
 // architecture-handbook-entry-end: Nodes
+
+message FatContractInstance {
+  bytes contract_id = 1;
+  com.daml.lf.value.Identifier template_id = 2;
+  bytes create_arg = 3;
+  bytes contract_key = 4;
+  repeated string maintainers = 5;
+  repeated string non_maintainer_signatories = 6;
+  repeated string non_signatory_stakeholders = 7;
+  int64 createTime = 8;
+  bytes canton_data = 9;
```

Comment:
maybe worth adding a comment on how this is expected to be used?

Reply:
Is this intended to hold the salt needed in calculating the calculation of a canton v2 suffix ? Will the engine create this or be provided with it ?

Reply:
Currently there is a builder that takes a create node, the effective time and the salt. 
mid term, my ojectives is to replace the create node by a `FatContractInstance`, canton will have to set the create time, the salt in the same time is suffixing the contract It before committing the transaction.